### PR TITLE
Make the banner route a dynamic property

### PR DESCRIPTION
**Pull request by brunojdo**
#### What is the purpose of this pull request?

Add a dynamic behavior to banner link. Now, it can be an internal route or an external link. 

#### Screenshots or example usage
https://bruno--storecomponents.myvtexdev.com/carousel
screen shot 2018-04-30 at 13 08 42

#### Types of changes
- [ ] Bug fix (the non-breaking change which fixes an issue)
- [x] New feature (the non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/Carousel.js [hunk 1]
```diff
@@ -65,21 +56,53 @@ class Carousel extends Component {
...
+        },
+        params: {
+          type: 'string',
+          description: 'Comma separated params, e.g.: key=value,a=b,c=d',
+          title: 'Params',
+        },
+      }
 
     const getRepeatedProperties = repetition =>
       keyBy(
```
Comment (salesfelipe) [inline react/Carousel.js]: Shouldn't it be lodash?

Comment (salesfelipe) [inline react/Carousel.js]: Shouldn't it be lodash?

Comment (brunojdo) [reply react/Carousel.js]: This discussion is still open.

Comment (salesfelipe) [reply react/Carousel.js]: 👍

File: react/Carousel.js [hunk 2]
```diff
@@ -194,7 +212,7 @@ class Carousel extends Component {
       <div className="vtex-carousel">
         {!loading && (
           <Slider {...settings}>
-            {banners.map(function (banner, i) {
+            {banners.map(function (banner) {
```
Comment (leticiawanderley) [inline react/Carousel.js]: use es6

File: react/Carousel.js [hunk 3]
```diff
@@ -66,20 +57,46 @@ class Carousel extends Component {
...
   }
-  static getSchema = ({ numberOfBanners, autoplay }) => {
-    numberOfBanners = numberOfBanners || 3
-    autoplay = autoplay || false
+
+  static getSchema = (props) => {
+    const numberOfBanners = props.numberOfBanners || 3
+    const autoplay = props.autoplay || false
+
+    /** Defines a internal route or external link for the Banner */
```
Comment (leticiawanderley) [inline react/Carousel.js]:
```diff
-a internal
+an internal
```

File: react/Carousel.js [hunk 4]
```diff
@@ -66,20 +57,46 @@ class Carousel extends Component {
...
+    const bannerLink = (typeOfRoute) =>
```
Comment (leticiawanderley) [inline react/Carousel.js]: Unnecessary parenthesis

File: react/Carousel.js [hunk 5]
```diff
@@ -66,20 +57,46 @@ class Carousel extends Component {
       },
     },
   }
-  static getSchema = ({ numberOfBanners, autoplay }) => {
-    numberOfBanners = numberOfBanners || 3
-    autoplay = autoplay || false
+
+  static getSchema = (props) => {
```
Comment (leticiawanderley) [inline react/Carousel.js]: Unnecessary parenthesis

**Pull request merged**